packages/blade/src/hooks/useExposeInputRef.ts

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import type { TextInput as TextInputReactNative } from 'react-native';
+
+type NativeHTMLInputRefProps = Pick<HTMLElement, 'focus' | 'scrollIntoView'>;
+
+/**
+ * A hook which only exposes the properties of html input element via imparative hook
+ *
+ * It avoids exposing other native properties of HTMLElement
+ * like `style` `classList` to avoid unintended usage of refs.
+ */
+const useExposeInputRef = (
+  targetRef: React.ForwardedRef<NativeHTMLInputRefProps>,
```

Comment:
why do we need 2 ref objects here as args?

Reply:
targetRef is the ref which will get exposed to the user. 

and inputRef is the one which has the actual internal ref to the html input element. If we just want to directly expose the ref we won't need to do this but since we are using useImparativeHandle hook we need it.

Reply:
no need to two args now since we create & return the inputRef.